src/libsync/propagateupload.cpp

```diff
@@ -171,18 +171,41 @@ void PropagateUploadFileCommon::setDeleteExisting(bool enabled)
 
 void PropagateUploadFileCommon::start()
 {
-    if (propagator()->account()->capabilities().clientSideEncryptionAvaliable()) {
-      _uploadEncryptedHelper = new PropagateUploadEncrypted(propagator(), _item);
-      connect(_uploadEncryptedHelper, &PropagateUploadEncrypted::folerNotEncrypted,
-        this, &PropagateUploadFileCommon::setupUnencryptedFile);
-      connect(_uploadEncryptedHelper, &PropagateUploadEncrypted::finalized,
-        this, &PropagateUploadFileCommon::setupEncryptedFile);
-      connect(_uploadEncryptedHelper, &PropagateUploadEncrypted::error,
-        []{ qCDebug(lcPropagateUpload) << "Error setting up encryption."; });
-      _uploadEncryptedHelper->start();
-   } else {
-      setupUnencryptedFile();
+    const auto rootPath = [=]() {
+        const auto result = propagator()->_remoteFolder;
+        if (result.startsWith('/')) {
+            return result.mid(1);
+        } else {
+            return result;
+        }
+    }();
+    const auto path = QString(rootPath + _item->_file);
+    const auto parentPath = path.left(path.lastIndexOf('/'));
```

Comment:
Looks nearly identical to line 346-355 in `propagatedownload.cpp`, maybe put into a helper to avoid duplication?

Reply:
Same found in `PropagateRemoteMkdir::slotMkdir()`

Reply:
Perhaps even more lines of the code, now that I took another look ;)

Reply:
Correct, and it's getting worse in the other PR I'm working on at the moment.

I'm aware of it and letting it slide on purpose for now, it looks like that as we take more scenarios into account there might be more variation of that code. Also I didn't find a proper place to put it in (yet) without inducing even more coupling.

So I won't address this just yet.